runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5653,18 +5654,20 @@ TR_J9ByteCodeIlGenerator::loadStatic(int32_t cpIndex)
          genTreeTop(treeTopNode);
          }
 
-      static char *disableFinalFieldFoldingInILGen = feGetEnv("TR_DisableFinalFieldFoldingInILGen");
-      static char *disableStaticFinalFieldFoldingInILGen = feGetEnv("TR_DisableStaticFinalFieldFoldingInILGen");
-      if (!disableFinalFieldFoldingInILGen &&
-          !disableStaticFinalFieldFoldingInILGen &&
-          symbol->isFinal() &&
-          TR::TransformUtil::canFoldStaticFinalField(comp(), load) == TR_yes)
-         {
-         TR::TransformUtil::foldReliableStaticFinalField(comp(), load);
-         }
-
       push(load);
       }
+
+   static char *disableFinalFieldFoldingInILGen = feGetEnv("TR_DisableFinalFieldFoldingInILGen");
+   static char *disableStaticFinalFieldFoldingInILGen = feGetEnv("TR_DisableStaticFinalFieldFoldingInILGen");
```

Comment:
```suggestion
   static bool disableFinalFieldFoldingInILGen = feGetEnv("TR_DisableFinalFieldFoldingInILGen") != NULL;
   static bool disableStaticFinalFieldFoldingInILGen = feGetEnv("TR_DisableStaticFinalFieldFoldingInILGen") != NULL;
```

We've ran the testing however so you don't need to make this change, but keep it in mind for future.

Reply:
For context, the reason I'm making this suggestion is because we want to move towards enabling warnings as errors, and that requires proper boolean comparison in `if` statements IIRC.

Reply:
Sorry, the PR was merged before I saw your comment, I'll keep this in mind and will change existing use when I can in my future PRs.